src/red.ts

```diff
@@ -129,36 +131,55 @@ export const serializedRedEnvSourceText = (function redEnvFactory(blueEnv: Membr
         if (typeof blue === 'function') {
             return getRedFunction(blue);
         }
-        let isBlueArray = false;
-        try {
-            isBlueArray = isArrayOrNotOrThrowForRevoked(blue);
-        } catch {
-            // blue was revoked - but we call createRedProxy to support distortions
-            return createRedProxy(blue);
-        }
-        if (isBlueArray) {
-            return getRedArray(blue);
-        } else if (typeof blue === 'object') {
+        if (typeof blue === 'object') {
```

Comment:
this is probably the most complicated logic in this file... need more eyeballs here.